src/openrct2/peep/Guest.cpp

```diff
@@ -1627,17 +1620,6 @@ bool Guest::DecideAndBuyItem(Ride* ride, int32_t shopItem, money32 price)
             HappinessTarget = std::min((HappinessTarget + happinessGrowth), PEEP_MAX_HAPPINESS);
             Happiness = std::min((Happiness + happinessGrowth), PEEP_MAX_HAPPINESS);
         }
-    }
-
-loc_69B221:
-    if (!hasVoucher)
-    {
-        if (gClimateCurrent.Temperature >= 21)
-            itemValue = ShopItems[shopItem].HotValue;
-        else if (gClimateCurrent.Temperature <= 11)
-            itemValue = ShopItems[shopItem].ColdValue;
-        else
-            itemValue = ShopItems[shopItem].BaseValue;
```

Comment:
I'm a bit confused by the removal of these. Sounds like it was used to reset the `itemValue`, but it isn't anymore,

Reply:
In the old code, it did the same operation twice. I removed the redundancy.

Reply:
But look in the old code it first reset `itemValue` depending on the temperature, then it seems like it always subtracted the price, so I'll call `itemValue - price` as `itemValueP`. Afterwards, it could do any number of operations. For the first if:
`[-itemValueP, -itemValueP/2, -itemValueP/4]`
For the else:
`[max(8, itemValueP)]`
For the second if, the one that calculates satisfaction, it reset back to `itemValueP` , but now it is whatever the end result of the if/else pair above ends up with, which in most of the cases isn't it (only 1 out of the 5 options). Does that make sense?

Reply:
Yes, I see your point.  I have restored the code.